### PR TITLE
feature: adding sudo call for linux

### PR DESCRIPTION
**Pull request by atma-stackoverflow**
Added sudo call for Linux, moved to go mod, and cleaned up windows file that breaks Linux build

### CODE REVIEW

**Comment by atma-stackoverflow** [comment]
```
$ ./pat
sudo: /opt/puppetlabs/bin/puppet: command not found
```

Sudo works, but this is on my personal system so the puppet command isn't found.

File: puppet_unix.go [hunk 1]
```diff
@@ -20,7 +20,8 @@ func isRoot() bool {
 
 // This is the normal way of execing, see the Windows one for the messed up way
 func osMakeExec(osPuppetBinPath string, puppetArgs ...string) (*exec.Cmd, error) {
-	cmd := exec.Command(osPuppetBinPath, puppetArgs...)
+	sudoArgs := append([]string{osPuppetBinPath}, puppetArgs...)
+	cmd := exec.Command("sudo", sudoArgs...)
```
Comment (tlimoncelli) [inline puppet_unix.go]: Suggestion:
Use the IsRoot() function to only add sudo if needed.

**Comment by tlimoncelli** [comment]
> ```
> $ ./pat
> sudo: /opt/puppetlabs/bin/puppet: command not found
> ```
> 
> Sudo works, but this is on my personal system so the puppet command isn't found.

Some organizations configure sudo to fail if you are already root. So if "sudo pat" tries to run "sudo /opt/puppetlabs/bin/puppet" they'd get an error message.

**Review by tlimoncelli — APPROVED** [review]
So much cleaner!  Great job!

**Pull request merged**